kymatio/scattering1d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,144 @@
+                    psi_f[sub_k] = torch.from_numpy(
+                        psi_f[sub_k]).float().view(-1, 1).repeat(1, 2)
+                    self.register_buffer('tensor' + str(n), psi_f[sub_k])
+                    n += 1
+        for psi_f in self.psi2_f:
+            for sub_k in psi_f.keys():
+                if type(sub_k) != str:
+                    # view(-1, 1).repeat(1, 2) because real numbers!
+                    psi_f[sub_k] = torch.from_numpy(
+                        psi_f[sub_k]).float().view(-1, 1).repeat(1, 2)
+                    self.register_buffer('tensor' + str(n), psi_f[sub_k])
+                    n += 1
+
+
+    def scattering(self, x):
+        """Apply the scattering transform
+
+        Given an input Tensor of size `(B, T0)`, where `B` is the batch
+        size and `T0` is the length of the individual signals, this function
+        computes its scattering transform. If the `vectorize` flag is set to
+        `True`, the output is in the form of a Tensor or size `(B, C, T1)`,
+        where `T1` is the signal length after subsampling to the scale `2**J`
+        (with the appropriate oversampling factor to reduce aliasing), and
+        `C` is the number of scattering coefficients.  If `vectorize` is set
+        `False`, however, the output is a dictionary containing `C` keys, each
+        a tuple whose length corresponds to the scattering order and whose
+        elements are the sequence of filter indices used.
+
+        Furthermore, if the `average` flag is set to `False`, these outputs
+        are not averaged, but are simply the wavelet modulus coefficients of
+        the filters.
+
+        Parameters
+        ----------
+        x : tensor
+            An input Tensor of size `(B, T0)`.
+
+        Returns
+        -------
+        S : tensor or dictionary
+            If the `vectorize` flag is `True`, the output is a Tensor
+            containing the scattering coefficients, while if `vectorize`
+            is `False`, it is a dictionary indexed by tuples of filter indices.
+        """
+        # basic checking, should be improved
+        if len(x.shape) < 1:
+            raise ValueError(
+                'Input tensor x should have at least one axis, got {}'.format(
+                    len(x.shape)))
+
+        batch_shape = x.shape[:-1]
+        signal_shape = x.shape[-1:]
+
+        x = x.reshape((-1, 1) + signal_shape)
+
+        # get the arguments before calling the scattering
+        # treat the arguments
+        if self.vectorize:
+            if not(self.average):
+                raise ValueError(
+                    'Options average=False and vectorize=True are ' +
+                    'mutually incompatible. Please set vectorize to False.')
+            size_scattering = precompute_size_scattering(
+                self.J, self.Q, max_order=self.max_order, detail=True)
+        else:
+            size_scattering = 0
+
+        n = 0
```

Comment:
Move the following code into a `load_filters` method, like in https://github.com/kymatio/kymatio/blob/eb5317c45d16e4de016088961cab7fa341758b6d/kymatio/scattering2d/frontend/torch_frontend.py#L49.